File: Backend/src/CareConnect/CareConnect.Infrastructure/Entities/Invoice.cs
using CareConnect.DTOs.Enums;
using CareConnect.Infrastructure.Entities.Common;

namespace CareConnect.Infrastructure.Entities;

public class Invoice : IAuditable
{
    public int Id { get; set; }

    public int ClientId { get; set; }

    public Client Client { get; set; } = null!;

    public int? VisitId { get; set; }

    public Visit? Visit { get; set; }

    public string InvoiceNumber { get; set; } = null!;

    public decimal Amount { get; set; }

    public InvoiceStatus Status { get; set; }

    public DateOnly IssuedDate { get; set; }

    public DateOnly? DueDate { get; set; }

    public DateOnly? PaidDate { get; set; }

    public DateTime CreatedAtUtc { get; set; }

    public DateTime? UpdatedAtUtc { get; set; }
}

File: Frontend/src/types/invoice.ts
import type { InvoiceStatus } from './enums'

/**
 * Mirrors CareConnect.Infrastructure.Entities.Invoice — there is no InvoiceDto, no controller,
 * and no AppService for it anywhere in the backend (only the EF entity + InvoiceConfiguration
 * exist), so there is currently nothing for an invoiceService to call. Modeled here so the shape
 * is ready the moment that endpoint exists; field names follow the same camelCase-over-JSON
 * convention every other type in this folder uses once such a DTO is added.
 */
export interface Invoice {
  id: number
  clientId: number
  visitId: number | null
  invoiceNumber: string
  amount: number
  status: InvoiceStatus
  issuedDate: string
  dueDate: string | null
  paidDate: string | null
}
